### DISCUSSION TITLE
Is There plan to support GraalVM Reachability Metadata

### DISCUSSION THREAD

**Discussion by chenliuyang1989** [paketo-buildpacks/feedback#61] [Ideas]
As Titled
Is There plan to support GraalVM Reachability Metadata，Thanks

**Comment by dmikusa** [comment]
@pivotal-david-osullivan has been talking with the Spring Boot team about this, at least supporting it for Spring Boot apps. @pivotal-david-osullivan what is the latest here?

For some background...

The problem is that GraalVM has implemented this as a part of their Gradle/Maven plugins, however, buildpacks run `native-image` directly. Given our buildpacks are written in Go and the GraalVM project is written in Java, we can't easily integrate with what they are doing. We'd have to either reimplement what they did or somehow call out to a Java-based CLI (which doesn't exist).

It is technically possible to run the Gradle/Maven plugin inside of the container as part of the build process, but the output of that is different from what the buildpacks currently expect and so it doesn't work. It outputs the binary, but the buildpacks expect a JAR and to then run `native-image` against that JAR.

If you have feedback on how you'd like to see this integrated, feel free to leave it below. Otherwise, it is on the roadmap, but it's not a straightforward problem so it might take a little time to work out.
